src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -50,11 +50,23 @@ trait Infer extends Checkable {
       && isVarArgTypes(formals1)
     )
     def lastType = formals1.last.dealiasWiden.typeArgs.head
-    def expanded(n: Int) = (1 to n).toList map (_ => lastType)
 
-    if (expandLast)
-      formals1.init ::: expanded(numArgs - numFormals + 1)
-    else
+    if (expandLast) {
+      // Optimized version of: formals1.init ::: expanded(numArgs - numFormals + 1)
+      val result = mutable.ListBuffer[Type]()
+      var fs = formals
+      while ((fs ne Nil) && (fs.tail ne Nil)) {
+        result.addOne(fs.head)
+        fs = fs.tail
+      }
+      var i = 0
+      val n = numArgs - numFormals + 1
+      while (i < n) {
+        result += lastType
```

Comment:
`lastType` doesn't change here but it's also a `def` (and calls `List#last)

Reply:
I've moved it to a `val` inside `if (expandLast)`